libs/api/database/src/lib/content/stores/content.store.ts

```diff
@@ -509,4 +509,20 @@ export class ContentStore {
     }
 
     //#endregion
+
+    //#region ---TAGS---
+
+    /**
+     * Go through every piece of content, and remove the given `tagId` from 
+     * its `tags` array.
+     * @param tagId The ID of the tag to remove.
+     */
+    public async removeTagReferences(tagId: string): Promise<void> {
+        await this.content.updateMany(
+            {},
+            { $pull: {tags: { _id: tagId }} }
+        );
```

Comment:
@Figments, please sanity check this logic, you're Mongoosier than I am

Reply:
this seems like it'd be INSANELY slow

Reply:
My understanding is that `updateMany` is implemented with an eye toward performance, and is notably faster than "update all of these things one by one".

In addition, I'm _hoping_ that tag deletion will be an extremely rare event, but something we should still support. If it's a moderator-only action, I feel like it's probably fine if performance sucks a little bit.

The biggest concern is how hard it'll nuke application performance while it's happening--that definitely feels like an issue in single-threaded Node-land.

Maybe we keep this around for now, while our dataset is still tiny, and consider replacing it with something smarter around the end of beta? We definitely _need_ to be able to delete tags, and leaving dangling tag IDs in works seems like a great way to create lots of work for us later =/